src/main/java/io/neonbee/NeonBee.java

```diff
@@ -241,7 +249,9 @@ public static NeonBee get(Vertx vertx) {
                 NeonBee neonBee = new NeonBee(vertx, options);
 
                 // load the configuration and boot it up, on failure close Vert.x
-                return neonBee.loadConfig().compose(config -> neonBee.boot()).recover(closeVertx).map(neonBee);
+                return neonBee.loadConfig().map(config -> config.setCompositeMeterRegistry(compositeMeterRegistry))
```

Comment:
Regarding point 1:
Unfortunately, we can't get rid of the setCompositeMeterRegistry method because we need the CompositeMeterRegistry in the VertxOptions and the NeonBeeConfig and it must be the same object.